### PR TITLE
closes #243, closes #443; add initial Github Actions CI support

### LINKED ISSUE #243: build automation with CI?
Github Actions & ~~Gitlab CI~~ provides a huge free credit theoretically enough for several distributions. Any plans to implement this?

### LINKED ISSUE #443: Automatic CI using wine-tkg-git
So some members of the grapejuice community (A Roblox on Linux community) including I are seeking to automatically compile wine with patches specific to the Roblox client. And we like using wine-tkg-git to do this since it facilitates the whole process really well, so I wanted to ask if it's possible to supply automatic input to the non-makepkg-build.sh script so we can automate the compile process or should I be doing something else?

### PR DESCRIPTION
**Pull request by paulcarroty**

### CODE REVIEW

File: .github/workflows/wine-arch.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Wine Archlinux CI
```
Comment (CodingKoopa) [inline .github/workflows/wine-arch.yml:1]:
```suggestion
name: Wine Arch Linux CI
```
See [here](https://wiki.archlinux.org/title/Arch_terminology#Arch_Linux).

File: .github/workflows/wine-arch.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    container: archlinux:latest
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Compilation
```
Comment (CodingKoopa) [inline .github/workflows/wine-arch.yml:17]: Names have inconsistent tenses - perhaps change this to `Compile`

File: .github/workflows/wine-arch.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        run: |
+          echo -e "[multilib]\nInclude = /etc/pacman.d/mirrorlist" >> /etc/pacman.conf
+          yes|pacman -Syu --noconfirm base-devel sudo
```
Comment (CodingKoopa) [inline .github/workflows/wine-arch.yml:20]: Is this `yes` necessary? I would expect `--noconfirm` to take care of that.

Comment (CodingKoopa) [inline .github/workflows/wine-arch.yml:20]: Does this actually work, without running `pacman-key` as described [here](https://hub.docker.com/_/archlinux)?

Comment (CodingKoopa) [inline .github/workflows/wine-arch.yml:20]: Are you sure `sudo` is necessary, even if we use `su` later on?

File: .github/workflows/wine-arch.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+          useradd user -G wheel && echo "user ALL=(ALL) NOPASSWD: ALL" >> /etc/sudoers
```
Comment (CodingKoopa) [inline .github/workflows/wine-arch.yml:21]: (ditto, about whether `/etc/sudoers` is necessary)

File: .github/workflows/wine-arch.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+          chown user -R . && cd wine-tkg-git
+          su user -c "yes|PKGDEST=/tmp/wine-tkg makepkg --noconfirm -si"
```
Comment (CodingKoopa) [inline .github/workflows/wine-arch.yml:23]: Unnecessary `yes`?

Comment (CodingKoopa) [inline .github/workflows/wine-arch.yml:23]: Unnecessary `--install`?

**Review by CodingKoopa — CHANGES_REQUESTED** [review]
Thanks! I added some initial notes on the Arch Linux CI.

File: .github/workflows/wine-arch.yml (hunk 3 above)
Comment (paulcarroty) [reply .github/workflows/wine-arch.yml:20]: `--noconfirm` failed several times, so added `yes` to be sure.

Comment (paulcarroty) [reply .github/workflows/wine-arch.yml:20]: It's working, I can add the key if it really necessary.

Comment (paulcarroty) [reply .github/workflows/wine-arch.yml:20]: Build scripts required it.

File: .github/workflows/wine-arch.yml (hunk 4 above)
Comment (paulcarroty) [reply .github/workflows/wine-arch.yml:21]: Non-root user required.

File: .github/workflows/wine-arch.yml (hunk 5 above)
Comment (paulcarroty) [reply .github/workflows/wine-arch.yml:23]: Doesn't works without `yes`.

Comment (paulcarroty) [reply .github/workflows/wine-arch.yml]: Fixed.

File: .github/workflows/wine-arch.yml (hunk 3 above)
Comment (CodingKoopa) [reply .github/workflows/wine-arch.yml:20]: Okay, as long as it works that should be okay.

Comment (CodingKoopa) [reply .github/workflows/wine-arch.yml:20]: Ah, I didn't think about that ^^

File: .github/workflows/wine-arch.yml (hunk 5 above)
Comment (paulcarroty) [reply .github/workflows/wine-arch.yml]: Deleted all `yes` in Arch CI.

Comment (CodingKoopa) [reply .github/workflows/wine-arch.yml]: I thought about it some more; this one is actually probably needed. I forgot that the `PKGBUILD` is interactive; you probably will need this particular one in order to select the default values for the TkG configuration.

**Comment by paulcarroty** [comment]
@CodingKoopa seems like all resolved, but got #237 issue with jack&jack2 on Arch.

File: .github/workflows/wine-arch.yml (hunk 5 above)
Comment (paulcarroty) [reply .github/workflows/wine-arch.yml]: Done.

File: .github/workflows/wine-ubuntu.yml [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      - name: Compilation
+        run: |
+          sudo dpkg --add-architecture i386 && sudo apt update
+          cd wine-tkg-git 
+          sed -i 's/distro=""/distro="debuntu"/' customization.cfg
+          yes|./non-makepkg-build.sh
+      - name: Archive the artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          name: wine-tkg-build
```
Comment (AdelKS) [inline .github/workflows/wine-ubuntu.yml:26]: You are capturing the wrong folder, the correct one is `non-makepkg-builds`. For now the artifacts are 3.4Gb x)

Comment (paulcarroty) [reply .github/workflows/wine-ubuntu.yml:26]: Fixed.

Comment (paulcarroty) [reply .github/workflows/wine-ubuntu.yml:26]: Your suggestion points to empty dir 😄

Comment (paulcarroty) [reply .github/workflows/wine-ubuntu.yml:26]: Fixed.

**Comment by Tk-Glitch** [comment]
Looks fine to me for its current purpose. There was a kinda unified route pending but I guess it could be simplified now with some additional work on linux-tkg prior. Thanks! Merging.
:frog: :heart:

**Comment by robigan** [comment]
Uhh big issue, I get a `fatal: could not create work tree dir 'community-patches': Permission denied` error when using community-patches flag

**Pull request merged**